resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -749,13 +749,52 @@ private[spark] object Config extends Logging {
       .checkValue(value => value > 0, "Gracefully shutdown period must be a positive time value")
       .createWithDefaultString("20s")
 
+  val KUBERNETES_INGRESS_ENABLED =
+    ConfigBuilder("spark.kubernetes.driver.ingress.enabled")
```

Comment:
I believe we don't need this because `KUBERNETES_INGRESS_HOST_PATTERN` is mandatory always. Shall we remove this redudant configuration?